### PR TITLE
Add missing parameter in C API

### LINKED ISSUE #11541: Missing argument in C API
## Overview
The C api (located in `db/c.cc` exposes the function: `rocksdb_options_add_compact_on_deletion_collector_factory`.

It is missing the argument `deletion_ratio` to be passed to the underlying object.

see function here:
https://github.com/facebook/rocksdb/blob/70bf5ef0934d46ec42870ee0869353fce086c12d/db/c.cc#L3883

## Details

The C API function `rocksdb_options_add_compact_on_deletion_collector_factory`
is missing the argument `deletion_ratio`. This argument can be passed to the underlying class factory `NewCompactOnDeletionCollectorFactory`.

This factory has a default value for that argument so it takes the default value and compilation passes with no error.
Default value is `0.0` or using this function we want to set a different value.

I can provide a PR to fix it if you are open to contributions.

### PR DESCRIPTION
**Pull request by lavigne958**
The class `NewCompactOnDeletionCollectorFactory` exposes the parameter `delete_ratio`.

The C API `rocksdb_options_add_compact_on_deletion_collector_factory` does not allow a user to pass a delete ration to be passed down the the C++ class bellow.

The class has default value for the delete ratio which makes it pass the compilation and the tests.

closes #11541

### CODE REVIEW

**Review by cbi42 — APPROVED** [review]
LGTM thanks. Can you also add release note by following `unreleased_history/README.txt`?

**Comment by lavigne958** [comment]
> unreleased_history/README.txt

Hi thank you, sure I can.

**Comment by lavigne958** [comment]
Can someone please re-run the flaky failing test in ci/circleci: build-linux-clang10-mini-tsan ?

File: unreleased_history/public_api_changes/add_new_compaction_parameter.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Expose delete ratio parameter from the compaction API to the C API
```
Comment (cbi42) [inline unreleased_history/public_api_changes/add_new_compaction_parameter.md:1]: Could you be more specific here, something like "Add parameter `deletion_ratio` to C API  `rocksdb_options_add_compact_on_deletion_collector_factory`"?

Comment (lavigne958) [reply unreleased_history/public_api_changes/add_new_compaction_parameter.md:1]: thanks for the restart, I updated the release note. let me know if anything is missing.

**Comment by lavigne958** [comment]
Should I rebase my branch on the main branch ? or I let you do it ? just merge like this ? :thinking:

**Comment by cbi42** [comment]
> Should I rebase my branch on the main branch ? or I let you do it ? just merge like this ? 🤔

Both are fine in this case, rebasing is probably safer in general.

**Pull request closed without merging**